app/services/iiif_metadata_service.rb

```diff
@@ -49,10 +49,13 @@ def retrieve
   end
 
   def json
+    retrieved_json = retrieve
     @json ||= begin
-                JSON.parse(retrieve).tap do |data|
+                JSON.parse(retrieved_json).tap do |data|
                   data['@id'] = @canonical_url
                 end
               end
+  rescue JSON::ParserError => error
+    raise Stacks::UnexpectedMetadataResponseError.new(@url, error, retrieved_json)
```

Comment:
I don't think we need to pass `error` as a parameter; it should be available in the exception as a `cause`, maybe?